nixops/nix_expr.py

```diff
@@ -1,55 +1,64 @@
 import functools
 import re
 import string
-from typing import Optional
+from typing import Any, Dict, List, Optional, Sequence, Tuple, TypeVar, Union, overload
 from textwrap import dedent
 
+import sys
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
 __all__ = ["py2nix", "nix2py", "nixmerge", "expand_dict", "RawValue", "Function"]
 
+Node = Union["MultiLineRawValue", "RawValue", "Container"]
+
 
 class RawValue(object):
-    def __init__(self, value):
+    def __init__(self, value: Any) -> None:
         self.value = value
 
-    def get_min_length(self):
+    def get_min_length(self) -> int:
         return len(self.value)
 
-    def is_inlineable(self):
+    def is_inlineable(self) -> bool:
         return True
 
-    def indent(self, level=0, inline=False, maxwidth=80):
+    def indent(self, level: int = 0, inline: bool = False, maxwidth: int = 80) -> str:
         return "  " * level + self.value
 
-    def __repr__(self):
+    def __repr__(self) -> str:
```

Comment:
I'm a bit unsure about this -- we say we're returning a str, but above we accept an Any in the constructor. What do you think?

Reply:
My understanding of Python is that `__repr__` should always return a string (and typeshed seems to agree). But yea, strange that it just returns the raw value and I don't know if the behavior is relied on somewhere :/